test/common/upstream/load_balancer_fuzz.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+package test.common.upstream;
+
+import "validate/validate.proto";
+import "envoy/config/cluster/v3/cluster.proto";
+import "google/protobuf/empty.proto";
+
+message UpdateHealthFlags {
+  // The host index determines what host set within the priority set which will get updated.
+  uint64 host_index = 1;
+  // These will determine how many hosts will get placed into health hosts, degraded hosts, and
+  // excluded hosts from the full host list.
+  uint32 num_healthy_hosts = 2;
+  uint32 num_degraded_hosts = 3;
+  uint32 num_excluded_hosts = 4;
+  // This is used to determine which hosts get marked as healthy, degraded, and excluded.
+  bytes random_bytestring = 5 [(validate.rules).bytes = {min_len: 1, max_len: 256}];
+}
+
+message LbAction {
+  oneof action_selector {
+    option (validate.required) = true;
+    UpdateHealthFlags update_health_flags = 1;
+    google.protobuf.Empty prefetch = 2;
+    google.protobuf.Empty choose_host = 3;
+  }
+}
+
+// This message represents what LoadBalancerFuzzBase will interact with, performing setup of host sets and calling into load balancers.
+// The logic that this message represents and the base class for load balancing fuzzing will be logic that maps to all types of load balancing
+// and can be used in a modular way at the highest level for each load balancer.
+message LoadBalancerTestCase {
+  envoy.config.cluster.v3.Cluster.CommonLbConfig common_lb_config = 1
+      [(validate.rules).message.required = true];
+  repeated LbAction actions = 2;
+
+  // TODO: Localities
+  repeated uint32 num_hosts_in_host_set = 3
+      [(validate.rules).repeated = {min_items: 1, max_items: 20}];
```

Comment:
nit: add comment. what does each of the number represent? 
suggest renaming to "num_hosts_in_priority_level"

Reply:
Renamed. Added comment.

Reply:
Like I said in comment below, changed "index" around my fuzzer to "priority"

Reply:
Where it's applicable :).